### ISSUE TITLE
foldStatement shadows Prelude.init

### ISSUE THREAD

**Issue by dustin** [nikita-volkov/hasql-th#27]
It looks like there's a symbol introduced that unnecessarily conflicts with `Prelude` and creates unnecessary warnings.

**Comment by nikita-volkov** [comment]
Can you elaborate please?

**Comment by dustin** [comment]
If you use `foldStatement` you'll probably get this error:

```
../gopro/src/GoPro/DB/Postgres.hs:161:30: warning: [-Wname-shadowing]
    This binding for ‘init’ shadows the existing binding
      imported from ‘Prelude’ at src/GoPro/DB/Postgres.hs:14:8-24
      (and originally defined in ‘GHC.List’)
    |       
161 | loadConfigS = [foldStatement|select key :: text, value :: text from config|] (Fold f mempty conv)
    |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I'm assuming that's this line:  https://github.com/nikita-volkov/hasql-th/blob/master/library/Hasql/TH/Construction/Exp.hs#L112

**Comment by nikita-volkov** [comment]
Addressed in `0.4.0.18`
